### PR TITLE
Provenance Tracking, Error Tracking

### PR DESCRIPTION
**Pull request by jlaura**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ install:
   - conda info -a
 
   # Create a virtual env and install dependencies
-  - conda create -y -q -n test-env python=$TRAVIS_PYTHON_VERSION nose numpy pillow scipy pandas networkx scikit-image sqlalchemy
+  - conda create -y -q -n test-env python=$TRAVIS_PYTHON_VERSION nose numpy pillow scipy pandas networkx scikit-image sqlalchemy numexpr
```
Comment (kree) [inline .travis.yml]: If you haven't already, could you also add this change to https://github.com/USGS-Astrogeology/autocnet/wiki/Development-Environment-Setup if necessary.

File: functional_tests/test_two_image.py [hunk 2]
```diff
@@ -67,21 +66,21 @@ def test_two_image(self):
...
+            self.assertIn(edge.masks[0]['ratio'].sum(), range(250, 350))
 
         # Step: Compute the homographies and apply RANSAC
         cg.compute_homographies(clean_keys=['symmetry', 'ratio'])
 
         # Step: Compute the overlap ratio and coverage ratio
-        for s, d, edge in cg.edges_iter(data=True):
-            ratio = edge.coverage_ratio(clean_keys=['symmetry', 'ratio'])
+        #for s, d, edge in cg.edges_iter(data=True):
+            #ratio = edge.coverage_ratio(clean_keys=['symmetry', 'ratio'])
```
Comment (kree) [inline functional_tests/test_two_image.py]: Why is this commented out now?

File: autocnet/matcher/outlier_detector.py [hunk 3]
```diff
@@ -176,12 +176,12 @@ def compute_homography(kp1, kp2, method='ransac', **kwargs):
...
-    outlier_algorithm : {'ransac', 'lmeds', 'normal'}
-                        The openCV algorithm to use for outlier detection
+    method : {'ransac', 'lmeds', 'normal'}
+             The openCV algorithm to use for outlier detection
 
-    reproj_threshold : float
-                       The maximum distances in pixels a reprojected points
-                       can be from the epipolar line to be considered an inlier
+    ransacReprojThreshold : float
+                            The maximum distances in pixels a reprojected points
```
Comment (kree) [inline autocnet/matcher/outlier_detector.py]: Why switch to camelCase?

Comment (jlaura) [inline autocnet/matcher/outlier_detector.py]: The parameter is passed through to OpenCV and OpenCV requires camel case - coming from a C++ API.

File: functional_tests/test_two_image.py (hunk 2 above)
Comment (jlaura) [inline functional_tests/test_two_image.py]: to get local tests passing with a bad gdal build - I will uncomment

File: autocnet/graph/tests/test_node.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        # Convex hull computation is checked lower in the hull computation
+        self.assertRaises(AttributeError, self.node.coverage_ratio)
+
+    def test_provenance(self):
+        image = self.node.get_array()
+        self.node.extract_features(image, extractor_parameters={'nfeatures':10})
+        self.node.extract_features(image, extractor_parameters={'nfeatures':15})
+        p0 = self.node.provenance[0]
+        p1 = self.node.provenance[1]
+        print(self.node.provenance)
```
Comment (kree) [inline autocnet/graph/tests/test_node.py]: print statement

**Comment by kree** [comment]
I'm merging this now! I had a couple of questions about things I saw on here that I asked about over on gitter, but they're questions about conventions not correctness :)

**Pull request merged**